plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -0,0 +1,353 @@
+
+
+@utf8_conversion
+@multiattr_conversion(
+    attr_conversion('epoch')(),
+    attr_conversion('version')(),
+    attr_conversion('release')()
+)
+def evr_unit_conversion(solvable, epoch, version, release):
+    if version is None:
+        return
+    return '{}{}{}'.format(
+        '{}:'.format(epoch) if epoch else '',
+        version,
+        '-{}'.format(release) if release else ''
+    )
+
+
+evr_attribute = setattr_conversion('evr')(evr_unit_conversion)
+
+
+@multiattr_conversion(
+    utf8_conversion(attr_conversion('name')()),
+    attr_conversion('flags')(),
+    evr_unit_conversion
+)
+def rpm_dependency_conversion(solvable, unit_name, unit_flags, unit_evr,
+                              attr_name, dependency_key=None):
+    """Set the solvable dependencies.
+
+    The dependencies of a unit are stored as a list of dictionaries,
+    containing following values:
+            name: <unit name> or a rich dep string; mandatory
+            version: version of the dependency; optional
+            epoch: epoch of the dependency; optional
+            release: release of the dependency; optional
+            flags: AND/OR; optional; if missing meaning by default AND
+
+    These values are parsed by librpm.
+    There are two cases how libsolv addresses the dependencies:
+
+    * rich: the name of the dependency contains all required information:
+        '(foo >= 1.0-3 AND bar != 0.9)'
+        all the other attribute values are ignored
+
+    * generic: the name, version, epoch, release and flags attributes
+        are processed explicitly
+
+    The dependency list is either of the provides, requires or the weak
+    dependencies, the current case being stored under self.attr_name.
+
+    Libsolv tracks a custom Dep object to represent a dependency of a
+    solvable object; these are created in the pool object:
+
+        dependency = pool.Dep('foo')
+
+    The relationship to the solvable is tracked by a Rel pool object:
+
+        relationship = pool.Rel(solv.REL_AND, pool.Dep(evr))
+
+    where the evr is the 'epoch:version-release' string. The relationship
+    is then recorded on the solvable explicitly by:
+
+        solvable.add_deparray(solv.SOLVABLE_PROVIDES, relationship)
+
+    If no explict relationship is provided in the flags attribute,
+    the dependency can be used directly:
+
+        solvable.add_deparray(solv.SOLVABLE_PROVIDES, dependency)
+
+    :param solvable: a libsolv solvable object
+    :type solvable: a libsolv solvable
+    :param unit: the content unit to get the dependencies from
+    :type unit: an object or a dictionary
+    :returns: None
+    """
+    # e.g SOLVABLE_PROVIDES, SOLVABLE_REQUIRES...
+    keyname = dependency_key or getattr(solv, 'SOLVABLE_{}'.format(attr_name.upper()))
+    pool = solvable.repo.pool
+    if unit_name.startswith('('):
+        # the Rich/Boolean dependencies have just the 'name' attribute
+        # this is always in the form: '(foo >= 1.2 AND bar != 0.9)'
```

Comment:
`(foo >= 1.2 with foo < 2.0)` should be better example

Reply:
right; will update